src/NuGet.Clients/NuGet.PackageManagement.UI/Options/AddMappingDialog.xaml.cs

```diff
@@ -0,0 +1,164 @@
+                    NuGetServices.SourceProviderService,
+                    cancellationToken: cancellationToken))
+            {
+                //show package sources on open
+                _originalPackageSources = await _nugetSourcesService.GetPackageSourcesAsync(cancellationToken);
+            }
+
+            SourcesCollection.Clear();
+            foreach (PackageSourceContextInfo source in _originalPackageSources)
+            {
+                var viewModel = new PackageSourceViewModel(source);
+                SourcesCollection.Add(viewModel);
+            }
+        }
+
+        private void ExecuteHideDialog(object parameter)
+        {
+            Close();
+        }
+
+        private void ExecuteAddMapping(object parameter)
+        {
+            if (string.IsNullOrWhiteSpace(_packageID.Text))
+            {
+                return;
+            }
+
+            string packageId = _packageID.Text;
+            List<PackageSourceContextInfo> packageSources = new List<PackageSourceContextInfo>();
+            foreach (PackageSourceViewModel source in _sourcesListBox.Items)
+            {
+                if (source.IsSelected)
+                {
+                    packageSources.Add(source.SourceInfo);
+                }
+            }
+            var viewModel = new SourceMappingViewModel(packageId, packageSources);
+            _parent.SourceMappingsCollection.Add(viewModel);
+
+            (_parent.ShowAddDialogCommand as DelegateCommand).RaiseCanExecuteChanged();
+            (_parent.RemoveMappingCommand as DelegateCommand).RaiseCanExecuteChanged();
+            (_parent.ClearMappingsCommand as DelegateCommand).RaiseCanExecuteChanged();
+            Close();
+        }
+
+        private bool CanExecuteAddMapping(object parameter)
+        {
+            if (string.IsNullOrWhiteSpace(_packageID.Text))
+            {
+                return false;
+            }
+
+            foreach (PackageSourceViewModel source in _sourcesListBox.Items)
+            {
+                if (source.IsSelected)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        // Allows the user to drag the window around
+        protected override void OnMouseLeftButtonDown(MouseButtonEventArgs e)
+        {
+            base.OnMouseLeftButtonDown(e);
+            DragMove();
+        }
+
+        private void CheckBox_Checked(object sender, RoutedEventArgs e)
+        {
+            var itemCheckBox = sender as CheckBox;
+            var itemContainer = itemCheckBox?.FindAncestor<ListViewItem>();
+            if (itemContainer is null)
+            {
+                return;
+            }
+
+            var newValue = (e.RoutedEvent == CheckBox.CheckedEvent);
+            var oldValue = !newValue; // Assume the state has actually toggled.
+            AutomationPeer peer = UIElementAutomationPeer.FromElement(itemContainer);
```

Comment:
Tested that Toggle pattern is working correctly for adding new mappings. Added `Newtonsoft.*` mapped to `nuget.org` entirely with the keyboard:

![image](https://user-images.githubusercontent.com/49205731/194180016-f21ff809-b052-4747-92c2-a436c097bb88.png)

![image](https://user-images.githubusercontent.com/49205731/194180112-2d03c664-aa95-4fd7-ab26-6a4af03afcc1.png)